### PR TITLE
Packet fixes and improvements

### PR DESCRIPTION
**Pull request by mulbc**
* Fix boot issues with `c2.medium` instance types
* Make it easier to chose facility
* Add usage files to .gitignore

### CODE REVIEW

File: upi/metal/main.tf [hunk 1]
```diff
@@ -98,14 +100,16 @@ resource "matchbox_group" "worker" {
 provider "packet" {}
 
 locals {
-  packet_facility = "sjc1"
+  # The packet facility to deploy to - core facilities:
+  # all,ams1,iad2,dc13,dfw2,ewr1,ny5,sin3,sjc1,nrt1
+  packet_facility = "any"
```
Comment (abhinavdahiya) [inline upi/metal/main.tf:105]: the nodes in the same cluster need to be in same facility because of network latency concerns.

File: upi/metal/main.tf [hunk 2]
```diff
@@ -8,13 +8,15 @@ provider "ignition" {
...
 locals {
+  initrd_list = split("/", var.pxe_initrd_url)
   kernel_args = [
     "console=tty0",
     "console=ttyS1,115200n8",
     "rd.neednet=1",
 
     # "rd.break=initqueue"
     "coreos.inst=yes",
+    "initrd=${ local.initrd_list[ length(local.initrd_list) - 1 ] }",
```
Comment (abhinavdahiya) [inline upi/metal/main.tf:19]: Can you explain a little why this change is required?

Comment (mulbc) [reply upi/metal/main.tf:19]: Without this, the c2.medium Packet instances would not boot, because they could not find the initramfs
https://gist.github.com/mulbc/eb4b82e5ea0d1c685e67f62af29db84e

With this, they booted just fine

File: upi/metal/main.tf (hunk 1 above)
Comment (mulbc) [reply upi/metal/main.tf:105]: Glad we agree here... which is why we should actually use this local variable to set the facility... right now master is set to provision the instances wherever, which can lead to masters, workers and bootstrap distributed around the world:

https://github.com/openshift/installer/blob/1da6208796da9e0fa4cd8a4fa1467094342c6168/upi/metal/main.tf#L108

https://github.com/openshift/installer/blob/1da6208796da9e0fa4cd8a4fa1467094342c6168/upi/metal/main.tf#L121

https://github.com/openshift/installer/blob/1da6208796da9e0fa4cd8a4fa1467094342c6168/upi/metal/main.tf#L146

Since I did not want to change "any" to something specific and break CI because there is no availability... I chose to change the variable to any for now - I can change it back to sjc1 if you think there's enough availability?

**Comment by mulbc** [comment]
Can someone have another look at this? I'm happy to do a rebase once we have cleared the discussions

File: upi/metal/main.tf (hunk 2 above)
Comment (abhinavdahiya) [reply upi/metal/main.tf:19]: we run CI tests with current configuration https://prow.ci.openshift.org/view/gcs/origin-ci-test/logs/release-openshift-ocp-installer-e2e-metal-4.6/1296164699951337472 and it seems to be working fine,

Can you explain a little bit more exactly what error msg / failure you are seeing.

File: upi/metal/main.tf (hunk 1 above)
Comment (abhinavdahiya) [reply upi/metal/main.tf:105]: let's use the local and keep it set to any...

File: .gitignore [hunk 3]
```diff
@@ -1 +1,3 @@
 /bin/
+upi/metal/matchbox/
+upi/metal/terraform.tfvars
```
Comment (abhinavdahiya) [inline .gitignore:3]: currently, we only keep the artifacts created by our build scripts in the .gitignore and therefore the upi/** changes should not be added to this file. see https://stackoverflow.com/questions/1753070/how-do-i-configure-git-to-ignore-some-files-locally for adding these to your local repo

Comment (mulbc) [reply .gitignore:3]: Ok I'll remove that change

File: upi/metal/main.tf (hunk 2 above)
Comment (mulbc) [reply upi/metal/main.tf:19]: Some packet instances don't not without this.
Since the Terraform code is hard coded to use small instances, you won't see the differences in the CI, but only when someone chooses different (larger) instances

File: upi/metal/main.tf [hunk 4]
```diff
@@ -98,14 +100,16 @@ resource "matchbox_group" "worker" {
...
+  # all,ams1,iad2,dc13,dfw2,ewr1,ny5,sin3,sjc1,nrt1
+  packet_facility = "any"
 }
 
 resource "packet_device" "masters" {
   count            = var.master_count
   hostname         = "master-${count.index}.${var.cluster_domain}"
   plan             = "c1.small.x86"
-  facilities       = ["any"]
+  facilities       = [local.packet_facility]
```
Comment (displague) [inline upi/metal/main.tf:112]: `local.packet_facility` could be kept as `any` if the `facilities` here (and on line 125) were extracted from the bootstrap node:

```
  facilities = [module.bootstrap.facility] # might need to add facility to the module outputs
```

Comment (mulbc) [reply upi/metal/main.tf:112]: indeed that would be another way to do this. But the current state of the tf can cause the deployment to be distributed across different facilities (e.g. bootstrap in a different facility as the workers and masters)

File: .gitignore (hunk 3 above)
Comment (mulbc) [reply .gitignore]: Removed 8d3bd1702cf751fadf5018408fa9761a005921b3 with that change

**Comment by mulbc** [comment]
Rebased to master so this can (finally?) be merged - please :)
